modules/core/src/main/java/org/apache/synapse/message/store/impl/rabbitmq/RabbitMQConsumer.java

```diff
@@ -34,255 +34,220 @@
-					             delivery.getProps().getMessageId());
-				}
-				return synapseMc;
-			}
-		} catch (ShutdownSignalException sse) {
-			logger.error(getId() + " connection error when receiving messages" + sse);
-		} catch (IOException ioe) {
-			logger.error(getId() + " connection error when receiving messages" + ioe);
-		}
-		return null;
-	}
+    /**
+     * Get a single message from the queue and deserialize for set into the message context
+     *
+     * @return the {@link MessageContext} with received message
+     */
+    @Override
+    public MessageContext receive() {
+        if (isAlive()) {
+            try {
+                GetResponse delivery = channel.basicGet(queueName, false);
+                if (delivery != null) {
+                    StorableMessage storableMessage = deserializeMessage(channel, delivery);
+                    org.apache.axis2.context.MessageContext axis2Mc = store.newAxis2Mc();
+                    MessageContext synapseMc = store.newSynapseMc(axis2Mc);
+                    synapseMc = MessageConverter.toMessageContext(storableMessage, axis2Mc, synapseMc);
+                    updateCache(delivery, delivery.getProps().getMessageId());
+                    log.info(getId() + " Received MessageId: " + delivery.getProps().getMessageId());
+                    if (log.isDebugEnabled()) {
+                        log.debug(getId() + " Received MessageId: " + delivery.getProps().getMessageId());
+                    }
+                    return synapseMc;
+                }
+            } catch (ShutdownSignalException | IOException e) {
+                log.error(getId() + " connection error when receiving messages." + e);
+            }
+        } else {
+            cleanup();
+            setConnection(store.createConnection());
+            setChannel(store.createChannel(connection));
+        }
+        return null;
+    }
 
-	public boolean ack() {
-		boolean result = cachedMessage.ack();
-		if (result) {
-			store.dequeued();
-		}
-		return result;
-	}
+    /**
+     * Deserialize the message to store in the queue
+     *
+     * @param channel  the {@link Channel} object to acknowledge the message in deserialization failure
+     * @param delivery the message received from the broker
+     * @return a {@link StorableMessage} object
+     * @throws IOException
+     */
+    private StorableMessage deserializeMessage(Channel channel, GetResponse delivery) throws IOException {
+        StorableMessage storableMessage = null;
+        try (ByteArrayInputStream inputStream = new ByteArrayInputStream(delivery.getBody());
+             ObjectInput objectInput = new ObjectInputStream(inputStream)) {
+            storableMessage = (StorableMessage) objectInput.readObject();
+        } catch (ClassNotFoundException e) {
+            log.error(getId() + "unable to read the stored message." + e);
+            channel.basicAck(delivery.getEnvelope().getDeliveryTag(), false);
```

Comment:
This is not something that needs to be handled by `deserializeMessage` method. Looking at the exception thrown, message consumer need to do this AFAIU.

Reply:
Ack. Move it to the `receive` method.